test/strategies/elkilpstrategy.js

```diff
@@ -0,0 +1,308 @@
+            await elkRouterContract.addLiquidity(wavaxTokenContract.address, elkTokenContract.address, await wavaxTokenContract.balanceOf(owner.address), await elkTokenContract.balanceOf(owner.address), 0, 0, owner.address, 1807909162115)
+        })
+
+        it('Withdraw correct amount of ELK Liquidity token from the strategy', async () => {
+            // We get the amount of ELK token we have
+            const amount = (await elkPairContract.balanceOf(owner.address)).toString();
+            // We get the 'totalDeposit' before we deposit into the strategy
+            const totalDepositsBefore = (await elkIlpStrategyV5.totalDeposits()).toString();
+
+            // We deposit into the staking contract our ELP tokens
+            await elkIlpStrategyV5.deposit(BigNumber.from((await elkPairContract.balanceOf(owner.address)).toString()))
+            // We get the 'totalDeposit' after our deposit into the strategy
+            const totalDepositsAfter = (await elkIlpStrategyV5.totalDeposits()).toString();
+
+            // We check all balances
+            expect((await elkIlpStrategyV5.balanceOf(owner.address)).toString()).to.equal(amount)
+            expect((await stakingContract.balanceOf(elkIlpStrategyV5.address)).toString()).to.equal(amount)
+            expect(BigNumber.from(totalDepositsAfter)).gt(BigNumber.from(totalDepositsBefore));
+            expect((await stakingContract.coverageOf(elkIlpStrategyV5.address)).toString()).to.equal("0")
+
+            // We impersonate the 'owner' of the WAVAX-ELK StakingRewardsILP contract
+            await ethers.provider.send('hardhat_impersonateAccount', ['0xba49776326A1ca54EB4F406C94Ae4e1ebE458E19']);
+            const admin = await ethers.provider.getSigner('0xba49776326A1ca54EB4F406C94Ae4e1ebE458E19')
+
+            const stakingcontract = await ethers.getContractAt(staking_contract, elpStakingRewardAddress, admin);
+            // We set the coverage amount for our Strategy
+            await stakingcontract.setCoverageAmount(elkIlpStrategyV5.address, 1000000000000);
+
+            await hre.network.provider.request({
+                method: "hardhat_stopImpersonatingAccount",
+                params: ["0xba49776326A1ca54EB4F406C94Ae4e1ebE458E19"],
+            });
+
+            // We check the amount of coverage the Strategy contract address currently has
+            expect((await stakingContract.coverageOf(elkIlpStrategyV5.address)).toString()).to.equal('1000000000000')
+            // expect((await elk_ilp_strategy_v5.checkCoverage()).toString()).to.equal('1000000000000')
+
+            // We should check now if, upon withdrawal, we get a bigger amount of ELP tokens
+            // We withdraw from the strategy all tokens we deposited
+            await elkIlpStrategyV5.withdraw(BigNumber.from((await elkIlpStrategyV5.balanceOf(owner.address)).toString()))
+            const totalDepositsAfterWithdraw = (await elkIlpStrategyV5.totalDeposits()).toString();
```

Comment:
I haven't completely understood what's the logic of checking the totalDeposits against the fact you got more ELK token/ One suggestion on how you can do this is actually use the hardhat time travelling feature, you can set to a specific time stamp, deposit, set to a new timestamp in the future, withdraw, check how much ELK you earned, then repeat this operation with the longest coverage streak(42 days), the latter should be bigger than the ELK received before

Reply:
Oh cool did not use that 'time travel' feature before , I'll check it .

I can only check the 'normal' reward with that mechanism, not the coverage. This is due to the fact that the coverage is distributed via : 
```solidity
    function setCoverageAmount(address addr, uint256 amount) public onlyOwner {
        _coverages[addr] = amount;
    }
```
So it needs an external call (which am doing via the 'impersonation' of the staking reward owner).